crates/red_knot/src/types/infer.rs

```diff
@@ -70,8 +70,27 @@ where
                 let ast = parsed.ast();
                 let node = node_key.resolve_unwrap(ast.as_any_node_ref());
 
+                let bases: Vec<_> = node
+                    .bases()
+                    .iter()
+                    .map(|base_expr| match base_expr {
+                        ast::Expr::Name(name) => {
+                            if let Some(base_symbol_id) = symbols.root_symbol_id_by_name(&name.id) {
+                                db.jar().type_store.record_symbol_dependency(
+                                    (file_id, symbol_id),
```

Comment:
I think I would prefer a named type instead of a two-tuple that records file and symbol id. `GlobalSymbolId`

Reply:
Yes, I planned to add this type, I should just do it in the previous PR already.

Reply:
I think dependency recording format needs more work, so I rebased these PRs without the dependency stuff yet, so this is gone now.